### ISSUE TITLE
"Type A is not assignable to type B." if union type has at least 26 members

### ISSUE THREAD

**Issue by zoltan-boros** [microsoft/TypeScript#51609]
# Bug Report

### 🔎 Search Terms

too many union members

### 🕗 Version & Regression Information

- This is a false type error: _"Type A is not assignable to type B. (2345)"_
- This is reproducible in all TS versions that are currently available in Playground (3.3.3 - 4.8.4, Nightly 5.0.0-dev.20221120)
- This is the behavior in every version I tried, and I reviewed the FAQ for entries about union types

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.8.4#code/C4TwDgpgBAKgDARigXigbygNwIYCcCW2AdsAFxQDkiFA3FAGb4QA2AJogPzkDOwBRAczoBfGgChQkWHABMKdFjyES5KjNoMmbWVyi9+QqKInho8AMzyMOAsTKU45jYxbtzu-fkEjxks3AAWK0VbFQcA5y12AI8+L0NjP2kAVmCbZXsqZMjXOGTYgx8TKXgANjSlO1U4UpztUoL4oqT4AHYK0My4Vrr2VsbvI19TaQAODozq0d64UYGE4ZK4AE4JqodlmeX55pGYBDg1sIoD3oOdoeKzBCRUa0rjm7OEC8S9hDk7kMnKD7OZV6La6WL7pdYnJx0FxsBDuHhxQZvEoIIKgh6ZFFnGLwwqXFoIVJozqqAlnfI4pp497lIk-E61KFRBANCmIoGwBDtWngzlnfqshZXDnjbmPaaM1wIOYC3bI1aijGbCUw7YyqklGSHBWqTW9TWAoUwGS3BRg47GvUvNVIswyT6m9E69TK1h2g0tGQgh3EyievVwvQIwUe1HeukyCIuiPuvYyQlh8FxvXkwO4m2wGQ0hPmhmaVyZg1CgCqRHwAHsiDARsgxFAoAAfaRIRvwOQtxwN6RBdupdvldvtdvjdurFsHWud-bNjltjmWMfdjm9jn9jmD4WThCjjNwCct42Tu2H+cZxdG5J7jPlAD016gAEl6FAQGWAK5QADGZYAtt+ICQoGAAALfBuCgZgvAgAAaQCgP-WDoCSCBcFwMtcCgbBgHA7BeHAyCoAEMsIDA7AAHdsBAAA6CdxDrMQvyIXDuB-CAADFXyID9gHLIh5AACjAPBsG-cgSx4qtIAAShQAA+BRBNwYSinoitcLNYBuFE0sKwkiAAG0KHUigAF09OM+QzPEMR1O4Sj6DQgBRbAPyAvj1NkvQWPYzjuIrPj7k6IxJMk8QgA)

### 💻 Code

```ts
type T01 = { variant: '01'; field01?: string; };
type T02 = { variant: '02'; field02?: string; };
type T03 = { variant: '03'; field03?: string; };
type T04 = { variant: '04'; field04?: string; };
type T05 = { variant: '05'; field05?: string; };
type T06 = { variant: '06'; field06?: string; };
type T07 = { variant: '07'; field07?: string; };
type T08 = { variant: '08'; field08?: string; };
type T09 = { variant: '09'; field09?: string; };
type T10 = { variant: '10'; field10?: string; };
type T11 = { variant: '11'; field11?: string; };
type T12 = { variant: '12'; field12?: string; };
type T13 = { variant: '13'; field13?: string; };
type T14 = { variant: '14'; field14?: string; };
type T15 = { variant: '15'; field15?: string; };
type T16 = { variant: '16'; field16?: string; };
type T17 = { variant: '17'; field17?: string; };
type T18 = { variant: '18'; field18?: string; };
type T19 = { variant: '19'; field19?: string; };
type T20 = { variant: '20'; field20?: string; };
type T21 = { variant: '21'; field21?: string; };
type T22 = { variant: '22'; field22?: string; };
type T23 = { variant: '23'; field23?: string; };
type T24 = { variant: '24'; field24?: string; };
type T25 = { variant: '25'; field25?: string; };
type T26 = { variant: '26'; field26?: string; };

type UnionType =
  | T01 | T02 | T03 | T04 | T05 | T06 | T07 | T08 | T09 | T10
  | T11 | T12 | T13 | T14 | T15 | T16 | T17 | T18 | T19 | T20
  | T21 | T22 | T23 | T24 | T25
  | T26 // If you comment this line, then the type error at last line goes away.
  ;
  
const someFunction = (param: UnionType) => { param; };

const variants: UnionType['variant'][] = [];

variants.forEach(variant => someFunction({ variant }));
```

### 🙁 Actual behavior

TS compiler produces false type error at last line:
_Type '"01"' is not assignable to type '"26"'. (2345)_

This behavior starts if `UnionType` has at least 26 members.

### 🙂 Expected behavior

TS can handle union types with more than 25 members.

**Comment by MartinJohns** [comment]
Duplicate of #32749 and others. Used search terms: `union 25 in:title`

**Comment by zoltan-boros** [comment]
Indeed, duplicate of #43283.
